riot_pal/serial_driver.py

```diff
@@ -48,11 +45,15 @@ def _connect(self, *args, **kwargs):
                     kwargs['port'] = os.environ['PORT']
                 else:
                     kwargs['port'] = self.DEFAULT_PORT
+
+        connect_wait = kwargs.pop('connect_wait', self.DEFAULT_CONNECT_WAIT)
         logging.debug("Serial connection args %r -- %r", args, kwargs)
         try:
             self._dev = Serial(*args, **kwargs)
         except SerialException:
             self._dev = serial_for_url(*args, **kwargs)
+        time.sleep(int(connect_wait))
```

Comment:
shouldn't this go above the `try ... except` block? I think creating the `Serial()` will also connect, or not, hence we should sleep before doing that?!

Reply:
The whole point of this is to sleep after connecting. This is the issue that the Arduino causes (connect resets then needs 3 seconds of silence to exit bootloader mode).  The other option would be to add a delay in every test then we won't need this.

Reply:
The whole point of this is to sleep after connecting. This is the issue that the Arduino causes (connect resets then needs 3 seconds of silence to exit bootloader mode). The other option would be to add a delay in every test then we won't need this.

Reply:
ah okay, never mind - if it works: good.